projects/libyaml/libyaml_dumper_fuzzer.c

```diff
@@ -24,9 +24,30 @@
 #undef NDEBUG
 #endif
 
-#define BUFFER_SIZE 65536
 #define MAX_DOCUMENTS 16
 
+typedef struct yaml_output_buffer {
+  unsigned char *buf;
+  size_t size;
+};
+
+static int yaml_write_handler(void *data, unsigned char *buffer, size_t size) {
+  struct yaml_output_buffer *out = (struct yaml_output_buffer *)data;
+  if (!out->size) {
```

Comment:
you don't need the first case, realloc(NULL, x) == malloc(x)

Reply:
Oh, I didn't know that! Thanks for the tip.